src/appmixer/mailerlite/core/CreateCampaign/CreateCampaign.js

```diff
@@ -0,0 +1,56 @@
+
+'use strict';
+
+module.exports = {
+    async receive(context) {
+
+        const { name, subject, senderName, fromAddress, content, groups } = context.messages.in.content;
+
+        if (!name) {
+            throw new context.CancelError('Campaign name is required!');
+        }
+        if (!subject) {
+            throw new context.CancelError('Email subject is required!');
+        }
+        if (!senderName) {
+            throw new context.CancelError('Email sender name is required!');
+        }
+        if (!fromAddress) {
+            throw new context.CancelError('Email from address is required!');
+        }
+        if (!content) {
+            throw new context.CancelError('Email content is required!');
+        }
+
+        const requestData = {
+            name: name,
+            type: 'regular',
+            emails: [
+                {
+                    subject: subject,
+                    from_name: senderName,
+                    from: fromAddress,
+                    content: content
+                }
+            ]
+        };
+
+        // Add optional fields
+        if (Array.isArray(groups?.AND) && groups.AND.length > 0) {
+            // Extract group IDs from the array of objects
+            requestData.groups = groups.AND.map(item => item.groups_item).filter(Boolean);
+        }
+
+        // https://developers.mailerlite.com/docs/#campaigns-create
+        const { data } = await context.httpRequest({
+            method: 'POST',
+            url: 'https://connect.mailerlite.com/api/campaigns',
+            headers: {
+                'Authorization': `Bearer ${context.auth.apiKey}`
+            },
+            data: requestData
+        });
+
+        return context.sendJson(data, 'out');
```

Comment:
I tried to run this component to see the response but I am getting error 422 "Sender email domain must be authenticated before you can send from this address." However, when looking into their documentation, I think the correct output should be from "data.data" (like in Create Subscriber).

```suggestion
        return context.sendJson(data.data, 'out');
```

Reply:
this should be data, I have updated the endpoint. Although it might give a 422 because our API key is not paid, so sending campaigns is not feasible